WPF/SeeShells/SeeShells/UI/EventParser.cs

```diff
@@ -0,0 +1,50 @@
+using SeeShells.ShellParser.ShellItems;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SeeShells.UI
+{
+    public class EventParser
+    {
+        /// <summary>
+        /// creates the eventList that will be returned full of events based on ShellItems
+        /// </summary>
+        List<IEvent> eventList = new List<IEvent>();
+        List<IShellItem> shells;
+        /// <summary>
+        /// Constructor for the EventParser that loops through each ShellItem and creates an event 
+        /// for each Modified,Accessed and Creation date for that shellItem.
+        /// </summary>
+        /// <param name="shells"></param>
+        public EventParser(List<IShellItem> shells)
+        {
+            this.shells = shells;
+        }
+        public List<IEvent> Parser()
+        {
+            foreach (IShellItem item in shells)
+            {
+                IDictionary<String, String> parser = item.GetAllProperties();
+                foreach(var el in parser)
+                {
+                    String check = el.Key;
+                    String date = "Date";
+                    if (check.Contains(date))
```

Comment:
This isnt the only check that should be performed
If you see a `DateTime.MinValue` (1/1/0001) you shouldnt make an event from this because it's a default value similar to `null` indicating there wasnt a real date.

```suggestion
                    if (check.Contains(date) && el.Value != DateTime.MinValue)
```

Reply:
oh got you makes sense